tests/system/Database/Live/InsertTest.php

```diff
@@ -92,6 +92,35 @@ public function testReplaceWithMatchingData()
         $this->assertSame('Cab Driver', $row->name);
     }
 
+    public function testReplaceTwice()
+    {
```

Comment:
I disagree. It is bad practice.

`testReplaceWithNoMatchingData()` is test for replace with no matching data.
`testReplaceWithMatchingData()` is test for replace with matching data.
`testReplaceTwice()` is test for replace twice.
They all differ in the intent of the test.

Reply:
If replace works twice then it also works once.

>testReplaceWithMatchingData() is test for replace with matching data.

Not sure what this is testing. It really doesn't matter if there is matching data or not. Whatever data is there is deleted and whatever is defined is replaced. Its not like it senses no changes and leaves it alone. If it works with mismatched data then is works without it.

I think we need to keep tests minimized where possible so that we don't have an hour of tests to run everytime. Yes they should be thorough though.

Reply:
You both have a good point.

I think we are used to making tests based on the behavior of the methods. In addition, if there is a bug, we always make a separate test for it (it's some sort of a rule). 

Even if merging everything into one test would save us 1 sec, I am not sure if it would be worth it in the long run. Readability of tests is important - we should not test 3 different things in one test.

I added a reference to the issue above the test and I'm gonna leave it as it is.

Reply:
Yes, slow tests are surely another issue.

> Not sure what this is testing. It really doesn't matter if there is matching data or not.

From a testing perspective, it really matters.
If you don't have the test, a needed test case is missing.

> Whatever data is there is deleted and whatever is defined is replaced.

We test to see if these data can really be replaced.
All those cases (with matching data or without matching data) should be tested.

> If replace works twice then it also works once.

It probably does, but not for sure. It depends on the production code.
But in this case, I also believe it does.